jobs/server.go

```diff
@@ -16,8 +18,9 @@ type JobServer struct {
 	ConfigService configservice.ConfigService
 	Store         store.Store
 	metrics       einterfaces.MetricsInterface
-	Workers       *Workers
-	Schedulers    *Schedulers
+	workers       *Workers
+	schedulers    *Schedulers
+	mut           sync.Mutex
```

Comment:
```suggestion
	workerSchedulerLock           sync.Mutex
```
... and/or maybe a comment about what one is meant to use it for?

Reply:
I assume that generally if not specified the mutex is used to protect access to the struct it's part of. I can rename it but if in the future we want to protect more fields it could get problematic. Adding a comment seems reasonable.

Reply:
It's pretty obvious that it's protecting something in the struct it's a member of. Can you tell what it's protecting without searching all the uses of `mut`: no you cannot. I have no preference whether that's resolved with a comment or a variable name.